eclair-core/src/main/scala/fr/acinq/eclair/db/pg/PgNetworkDb.scala

```diff
@@ -65,30 +66,35 @@ class PgNetworkDb(implicit ds: DataSource) extends NetworkDb with Logging {
         statement.executeUpdate("ALTER TABLE pruned_channels SET SCHEMA network")
       }
 
+      def migration45(statement: Statement): Unit = {
+        // We add channel updates to the pruned table.
+        statement.executeUpdate("ALTER TABLE network.pruned_channels ADD COLUMN channel_update_1 BYTEA NULL")
+        statement.executeUpdate("ALTER TABLE network.pruned_channels ADD COLUMN channel_update_2 BYTEA NULL")
+        // We clean up channels that contain an invalid channel update (e.g. missing htlc_maximum_msat).
+        statement.executeQuery("SELECT short_channel_id, channel_update_1, channel_update_2 FROM network.public_channels").map(rs => {
+          val shortChannelId = rs.getLong("short_channel_id")
+          val validChannelUpdate1 = rs.getBitVectorOpt("channel_update_1").forall(channelUpdateCodec.decode(_).isSuccessful)
+          val validChannelUpdate2 = rs.getBitVectorOpt("channel_update_2").forall(channelUpdateCodec.decode(_).isSuccessful)
+          (shortChannelId, validChannelUpdate1 && validChannelUpdate2)
+        }).collect {
+          case (scid, false) =>
+            logger.warn(s"removing channel update with scid=$scid from the network DB (update cannot be decoded)")
+            statement.executeUpdate(s"DELETE FROM network.public_channels WHERE short_channel_id=$scid")
+        }
+      }
+
       getVersion(statement, DB_NAME) match {
         case None =>
           statement.executeUpdate("CREATE SCHEMA network")
           statement.executeUpdate("CREATE TABLE network.nodes (node_id TEXT NOT NULL PRIMARY KEY, data BYTEA NOT NULL, json JSONB NOT NULL)")
           statement.executeUpdate("CREATE TABLE network.public_channels (short_channel_id BIGINT NOT NULL PRIMARY KEY, txid TEXT NOT NULL, channel_announcement BYTEA NOT NULL, capacity_sat BIGINT NOT NULL, channel_update_1 BYTEA NULL, channel_update_2 BYTEA NULL, channel_announcement_json JSONB NOT NULL, channel_update_1_json JSONB NULL, channel_update_2_json JSONB NULL)")
-          statement.executeUpdate("CREATE TABLE network.pruned_channels (short_channel_id BIGINT NOT NULL PRIMARY KEY)")
-        case Some(v@(2 | 3)) =>
+          statement.executeUpdate("CREATE TABLE network.pruned_channels (short_channel_id BIGINT NOT NULL PRIMARY KEY, channel_update_1 BYTEA NULL, channel_update_2 BYTEA NULL)")
```

Comment:
This table is never cleaned up, because on restart we only put `WatchExternalChannelSpent` on non-pruned channels.

An alternative would be to get rid of the `pruned_channels ` table, and keep pruned channels in the `public_channels` table. We could add an additional `is_pruned` flag in that table, or possibly even not store a flag at all and compute the _staleness_ status at runtime. Leaving stale channels in the `public_channels` table makes sense because they do still exist, and have already been validated. It also opens up the possibility to quickly revive a channel to non-stale by just updating its `channel_update`, without requesting announcements from our peer.